env/env.go

```diff
@@ -26,9 +27,15 @@ type Source struct {
 // it to UPPER_SNAKE_CASE. (The casing of `dials_env` and `dials` tags is left
 // unchanged.)
 func (e *Source) Value(t *dials.Type) (reflect.Value, error) {
-	stringCastingMangler := &transform.StringCastingMangler{}
+	// copy tags from "dials" to "dials_env" tag
 	tagCopyingMangler := &tagformat.TagCopyingMangler{SrcTag: transform.DialsTagName, NewTag: envTagName}
-	tfmr := transform.NewTransformer(t.Type(), stringCastingMangler, tagCopyingMangler)
+	// flatten the nested fields
+	flattenMangler := transform.NewFlattenMangler(transform.DialsTagName, caseconversion.EncodeUpperCamelCase, caseconversion.EncodeUpperCamelCase)
+	// reformat the tags so they are SCREAMING_SNAKE_CASE
+	reformatTagMangler := tagformat.NewTagReformattingMangler(transform.DialsTagName, caseconversion.DecodeGolangCamelCase, caseconversion.EncodeUpperSnakeCase)
+	// convert all the fields in the flattened struct to string type the environment variables can be set
+	stringCastingMangler := &transform.StringCastingMangler{}
+	tfmr := transform.NewTransformer(t.Type(), tagCopyingMangler, flattenMangler, reformatTagMangler, stringCastingMangler)
```

Comment:
Hmm, I think `tagCopyingMangler` should go last so the final value of the `dials_env` tag comes from the `dials` tag after we've flattened everything. (or at least after `flattenMangler` and `reformatTagMangler`)